docs/design/vcpkg.md

```diff
@@ -0,0 +1,159 @@
+dependencies. Ports can carry patches to be applied to sources of packages.
+
+<details>
+  <summary>A typical vcpkg.json</summary>
+
+  ```
+  {
+  "name": "fmt",
+  "version": "11.0.2",
+  "port-version": 1,
+  "description": "{fmt} is an open-source formatting library providing a fast and safe alternative to C stdio and C++ iostreams.",
+  "homepage": "https://github.com/fmtlib/fmt",
+  "license": "MIT",
+  "dependencies": [
+      {
+        "name": "vcpkg-cmake",
+        "host": true
+      },
+      {
+        "name": "vcpkg-cmake-config",
+        "host": true
+      }
+    ]
+  }
+  ```
+
+</details>
+
+
+<details>
+  <summary>The relevant part of corresponding portfile.cmake</summary>
+
+  ```
+  vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO fmtlib/fmt
+    REF "${VERSION}"
+    SHA512 47ff6d289dcc22681eea6da465b0348172921e7cafff8fd57a1540d3232cc6b53250a4625c954ee0944c87963b17680ecbc3ea123e43c2c822efe0dc6fa6cef3
+    HEAD_REF master
+    PATCHES
+        fix-write-batch.patch
+        fix-pass-utf-8-only-if-the-compiler-is-MSVC-at-build.patch # remove in next release
+  )
+  ...
+  ```
+
+</details>
+
+Dependencies come from registries which could be of three types:
+
+ 1. The "built-in" registry (effectively a local copy of [vcpkg][] repository);
+ 1. A git registry -- any repository which follows certain layout rules and
+    contains ports;
+ 1. A directory maintaining certain structure.
+
+
+## Downloading dependencies
+
+vcpkg would work on a project created with it. Before using the tool for a new
+project [the official guide][] recommends acquiring vcpkg repository and
+pointing an environment variable to it. This location (`$VCPKG_ROOT`) will be
+used to store sources for dependencies. There is no separate command for just
+downloading sources, however `vcpkg install` accepts `--only-downloads` flag
+which does exactly that. When sources are downloaded they modify two locations:
+vcpkg root directory (effectively a location to which [vcpkg][] repository was
+cloned) and registry cache. The latter defaults to `$HOME/.cache/vcpkg/registries/`
+but could be retargeted with `$X_VCPKG_REGISTRIES_CACHE`. Once both caches are
+populated a package could be built by running `vcpkg install --no-downloads`.
+Setting registries cache apparently always results in re-downloading a registry
+(unless there is a `--no-downloads` flag present).  Note, that if any patches
+are present in a port they will be applied to sources at this time. Transitive
+dependencies appear to be handled as well.
+
+In case of necessity dependency graph could be computed and represented in a
+variety of formats (excluding json) by running `vcpkg depend-info fmt`.
+
+
+## Proposed solution
+
+Following the precedent set in other package managers vcpkg should be used as
+an external tool. It will need to be downloaded and bootstrapped first, then
+caches locations will need to be set, after which package's dependencies
+could be resolved.
```

Comment:
So, since `vcpkg` is OSS (kudos Microsoft!), can we have a look at their implementation for inspiration, see what it is they do with `vcpkg install --only-download` and assess if reimplementing the fetching logic is totally impractical from our POV?

Reply:
We would need to not only re-implement fetch logic, but also provide correct repository structure for a vcpkg instance down the road. As always -- this is possible, but not very practical since we would need to maintain it. And since vcpkg relies heavily on cmake wouldn't that also mean that we would need to re-implement cmake functionality?